src/tasks.rs

```diff
@@ -86,5 +86,15 @@ pub extern "C" fn worker_task2() {
 
         #[cfg(feature = "wifi")]
         crate::wifi::send_data_if_needed();
+
+        // ideally we would yield instead of wait here (https://github.com/esp-rs/esp-wifi/issues/10)
+        let target = get_systimer_count() + (crate::timer::TICKS_PER_SECOND / 1000);
```

Comment:
This can fail if current time + target overflows the 54bits of the systimer. Writing it like we do in [xtensa_lx](https://github.com/esp-rs/xtensa-lx/blob/3e2120aaea0e83dd6e949cdaa24d16edd07632c8/src/timer.rs#L88) should prevent that from happening.

Reply:
good catch!

Reply:
This was my only review comment as well, going to verify on hardware.